scripts/README_VERIFICATION.md

```diff
@@ -0,0 +1,171 @@
+# Enhanced Contract Verification System
+
+The `verify.py` script has been enhanced to provide robust retry mechanisms and support for multiple block explorers while maintaining the same simple interface.
+
+## Overview
+
+The verification system has been enhanced to address the following requirements:
+
+1. **Multiple Block Explorer Support**: Verify contracts on at least 2 well-known block explorers
+2. **Robust Retry Mechanisms**: Implement proper retry logic with exponential backoff and jitter
+3. **Fallback Options**: If one verifier fails, automatically try others
+4. **Backward Compatibility**: Maintain compatibility with existing Makefile commands
+
+## Key Features
+
+### **Retry Logic with Exponential Backoff**
+```python
+@retry_with_backoff(max_retries=3, base_delay=2, max_delay=60)
+def api_request():
+    # Automatic retry with exponential backoff + jitter
+```
+
+### **Multi-Verifier Support**
+- **Etherscan**: Primary verifier with full API support
+- **Sourcify**: Secondary verifier (no API key required)
+- **Automatic Fallback**: If one fails, tries the next
```

Comment:
I believe it would be more beneficial to change this logic:

- verify on both by default.
- the script succeeds if any of the 2 verifications go through.